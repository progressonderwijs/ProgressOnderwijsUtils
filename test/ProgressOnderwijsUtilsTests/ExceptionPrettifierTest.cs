using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using ProgressOnderwijsUtils.ErrorHandling;
using ProgressOnderwijsUtils.Test;

namespace ProgressOnderwijsUtilsTests
{
	public class ExampleTestClass
	{
		[MethodImpl(MethodImplOptions.NoInlining)]
		public static void CausesError()
		{
			throw new Exception("This is an exception");
		}

		[MethodImpl(MethodImplOptions.NoInlining)]
		public void IndirectErrorViaInterface<T>(T param, string arg2 = "bla")
		{
			((IExampleTestInterface)new NestedClass()).SomeMethod();
		}


		interface IExampleTestInterface
		{
			void SomeMethod();
		}


		class NestedClass : IExampleTestInterface
		{
			void IExampleTestInterface.SomeMethod() { throw new NotImplementedException(); }
		}
	}


	[UseReporter(typeof(DiffReporter))]
	public class ExceptionPrettifierTest
	{
		[Test, MethodImpl(MethodImplOptions.NoInlining), Continuous]
		public void TrivialStackTraceWorks()
		{
			try
			{
				ExampleTestClass.CausesError();
			}
			catch (Exception e)
			{
				Approvals.Verify(
					ExceptionPrettifier.PrettyPrintException(e));
			}
		}

		[Test, MethodImpl(MethodImplOptions.NoInlining), Continuous]
		public void ExplicitInterfaceImplementationInNestedClass()
		{
			try
			{
				new ExampleTestClass().IndirectErrorViaInterface(42);
			}
			catch (Exception e)
			{
				Approvals.Verify(
					ExceptionPrettifier.PrettyPrintException(e));
			}
		}
	}
}