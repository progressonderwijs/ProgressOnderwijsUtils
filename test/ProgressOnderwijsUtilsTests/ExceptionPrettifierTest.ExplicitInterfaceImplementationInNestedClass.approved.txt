The method or operation is not implemented.
   at Progress.Test.Tools.ExampleTestClass.NestedClass.SomeMethod() in c:\hg\progress\test\Tools\ExceptionPrettifierTest.cs:line 35
   at Progress.Test.Tools.ExampleTestClass.IndirectErrorViaInterface<T>(T param, string arg2) in c:\hg\progress\test\Tools\ExceptionPrettifierTest.cs:line 23
   at Progress.Test.Tools.ExceptionPrettifierTest.ExplicitInterfaceImplementationInNestedClass() in c:\hg\progress\test\Tools\ExceptionPrettifierTest.cs:line 62
